tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,335 @@
+            previous_num=initial_num_of_pods, namespace=namespace,
+            change_type='decrease', min_difference=1, timeout=30, interval=0.01
+        )
+
+        assert pvc_deleting.result(), (
+            "Wait timeout: PVCs are not being deleted."
+        )
+        logging.info(
+            f"PVCs deletion has started."
+        )
+
+        assert pod_deleting.result(), (
+            "Wait timeout: Pods are not being deleted."
+        )
+        logging.info(
+            f"Pods deletion has started."
+        )
+
+        # Delete pod of type 'resource_to_delete'
+        disruption.delete_resource()
+
+        pods_deleted = pod_bulk_delete.result()
+        assert pods_deleted, "Deletion of pods failed."
+
+        # Verify pods are deleted
+        for pod_obj in pods_to_delete:
+            assert pod_obj.ocp.wait_for_delete(pod_obj.name), (
+                f"Pod {pod_obj.name} is not deleted"
+            )
+        logging.info("Verified: Pods are deleted.")
+
+        # Verify that the mount point is removed from nodes after deleting pod
+        node_pv_mounted = verify_pv_mounted_on_node(node_pv_dict)
+        for node, pvs in node_pv_mounted.items():
+            assert not pvs, (
+                f"PVs {pvs} is still present on node {node} after "
+                f"deleting the pods."
+            )
+        log.info(
+            "Verified: mount points are removed from nodes after deleting "
+            "the pods"
+        )
+
+        pvcs_deleted = pvc_bulk_delete.result()
+        assert pvcs_deleted, "Deletion of PVCs failed."
+
+        # Verify PVCs are deleted
+        for pvc_obj in pvcs_to_delete:
+            assert pvc_obj.ocp.wait_for_delete(pvc_obj.name), (
+                f"PVC {pvc_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVCs are deleted.")
+
+        # Verify PVs are deleted
+        for pv_obj in pv_objs:
+            assert pv_obj.ocp.wait_for_delete(
+                resource_name=pv_obj.name, timeout=300
+            ), (
+                f"PV {pv_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVs are deleted.")
+
+        # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        for pvc_name, uuid in pvc_uuid_map.items():
+            if interface == constants.CEPHBLOCKPOOL:
+                ret = verify_volume_deleted_in_backend(
+                    interface=interface, image_uuid=uuid,
+                    pool_name=sc_obj.ceph_pool.name
+                )
+            if interface == constants.CEPHFILESYSTEM:
+                ret = verify_volume_deleted_in_backend(
+                    interface=interface, image_uuid=uuid
+                )
+            assert ret, (
+                f"Volume associated with PVC {pvc_name} still exists "
+                f"in backend"
+            )
+
+        log.info("Fetching IO results from the pods.")
+        for pod_obj in io_pods:
+            get_fio_rw_iops(pod_obj)
```

Comment:
we need to have some check here on the returned data

Reply:
we need to understand if we actually managed to run FIO or not

Reply:
Fio verification added